sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -350,6 +350,9 @@ object QueryTest {
     Row.fromSeq(row.toSeq.map {
       case null => null
       case d: java.math.BigDecimal => BigDecimal(d)
+      // Round up the values to avoid approximate error
+      case f : Float => BigDecimal(f).setScale(8, BigDecimal.RoundingMode.HALF_UP).toFloat
```

Comment:
In your case, the difference between 2 numbers is too large to be considered as same, The 2 value should be compared in its absolute value, not relative. The float error won't be that large in the real world, and it should fail the test.